src/components/VideoVisualizer/VideoVisualizer.component.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { FaHeart } from 'react-icons/fa';
+import {
+  RelatedVideosContainer,
+  RelatedVideosTitle,
+  VideoPlayer,
+  VideoVisualizerContainer,
+} from './VideoVisualizer.elements';
+import { CardVideo } from '../CardVideo/CardVideo.component';
+import VideoContext from '../Context/VideoContext';
+import { useFavorites } from '../../providers/Favorites';
+
+function VideoVisualizer({ videoId, videos }) {
+  const [listVideos, setListVideo] = useState();
+  const { video } = useContext(VideoContext);
+  useEffect(() => {
+    if (videos) {
+      setListVideo(
+        videos.map((videoRelated) => (
+          <CardVideo
+            id={videoRelated.id.videoId}
+            key={videoRelated.etag}
+            video={videoRelated}
+          />
+        ))
+      );
+    }
+  }, [videos]);
+
+  const [buttonClicked, setButtonClicked] = useState(false);
+  const [state, dispatch] = useFavorites();
+  console.log(state);
+  const handleFavClick = () => {
+    if (buttonClicked) {
+      dispatch({
+        type: 'DELETE',
+        payload: video,
+      });
+      setButtonClicked(false);
+    } else {
+      dispatch({
```

Comment:
As an optional challenge, try to move the dispatch calls to inside the `FavoritesProvider` so that way instead of exposing `dispatch` and need to remember the exact `type` of action, you expose more straightforward functions like `addFavorite` and `deleteFavorite` that take the payload as the only parameter